### PR TITLE
plot: add autogenerated coordinates

### LINKED ISSUE #111: network plot automated layout if no coordinates given

### PR DESCRIPTION
**Pull request by fneum**
As requested in https://groups.google.com/forum/#!topic/pypsa/kEYy073p7ko, this PR adds functionality to `pypsa.plot` to automatically generate coordinates if `network.buses[['x','y']]` are missing or are to be overruled by a `networkx` layouting function through the `layouter` argument (closes #111). Defaults to reasonable layouts: planar if possible and kamadai-kawai otherwise.

```rst
    layouter : networkx.drawing.layout function, default None
        Layouting function from `networkx <https://networkx.github.io/>`_ which
        overrules coordinates given in ``network.buses[['x','y']]``. See
        `list <https://networkx.github.io/documentation/stable/reference/drawing.html#module-networkx.drawing.layout>`_
        of available options.
```

It is also possible to call `pypsa.plot.autogenerate_coordinates(network, assign=True)` outside the plotting function to write the coordinates into `network.buses[['x','y']]`.

Screenshot from 2019-12-22 16-59-52

### CODE REVIEW

File: pypsa/plot.py [hunk 1]
```diff
@@ -543,6 +548,61 @@ def directed_flow(n, flow, x=None, y=None, ax=None, geomap=True,
...
+    coordinates = pd.DataFrame(layouter(G)).T.rename({0: 'x', 1: 'y'}, axis=1)
+    
+    if assign:
+        network.buses[['x', 'y']] = coordinates
+    
+    return coordinates
+
+
+def _get_coordinates(network, layouter=None):
+    if layouter is not None or network.buses[['x', 'y']].isin([0]).all().all():
```
Comment (FabianHofmann) [inline pypsa/plot.py]: if layouter is not None or network.buses[['x', 'y']].isin([np.nan, 0]).all().all():

File: pypsa/plot.py [hunk 2]
```diff
@@ -32,8 +32,8 @@
 logger = logging.getLogger(__name__)
 
 
-__author__ = "Tom Brown (FIAS), Jonas Hoersch (FIAS)"
-__copyright__ = "Copyright 2015-2017 Tom Brown (FIAS), Jonas Hoersch (FIAS), GNU GPL 3"
+__author__ = "Tom Brown (FIAS), Jonas Hoersch (FIAS), Fabian Hofmann (FIAS), Fabian Neumann (KIT)"
+__copyright__ = "Copyright 2015-2019 Tom Brown (FIAS), Jonas Hoersch (FIAS); Copyright 2019 Fabian Hofmann (FIAS), Fabian Neumann (KIT), GNU GPL 3"
```
Comment (FabianHofmann) [inline pypsa/plot.py]:
```suggestion
__copyright__ = "Copyright 2015-2020 Tom Brown (FIAS), Jonas Hoersch (FIAS); Copyright 2019 Fabian Hofmann (FIAS), Fabian Neumann (KIT), GNU GPL 3"
```
;)

Comment (fneum) [reply pypsa/plot.py]: We can also make this complete ;)
```suggestion
__copyright__ = "Copyright 2015-2020 Tom Brown (FIAS), Jonas Hoersch (FIAS); Copyright 2019-2020 Fabian Hofmann (FIAS), Fabian Neumann (KIT), GNU GPL 3"
```

File: pypsa/plot.py [hunk 3]
```diff
@@ -544,19 +549,74 @@ def directed_flow(n, flow, x=None, y=None, ax=None, geomap=True,
...
+
 
 #This function was borne out of a breakout group at the October 2017
 #Munich Open Energy Modelling Initiative Workshop to hack together a
 #working example of plotly for networks, see:
 #https://forum.openmod-initiative.org/t/breakout-group-on-visualising-networks-with-plotly/384/7
 
 #We thank Bryn Pickering for holding the tutorial on plotly which
-#inspired the breakout group and for contributing ideas to the iplot
+#inspired the breakout group and for contributing ideas to thelayouter=None,  iplot
```
Comment (fneum) [inline pypsa/plot.py]: Ups...
```suggestion
#inspired the breakout group and for contributing ideas to the iplot
```

**Pull request merged**